pkgs/rbrowser/default.nix

```
@@ -1,202 +1,262 @@
-    #  along with this program; if not, write to the Free Software
-    #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
-    #  USA.
-    #
-
-    set -euo pipefail
-
-    function help()
-    {
-      echo -e "USAGE: rbrowser [options] <url>"
-      echo -e "\t -h, --help                - this message"
-      echo -e "\t -p, --profile             - the profile to use"
-      echo -e "\t -b, --browser             - the browser to use. Default: chromium"
-    }
-
-    ${optionalString withBrave ''brave="${brave}/bin/brave"''}
-    ${optionalString withChromium ''chromium="${chromium}/bin/chromium"''}
-    ${optionalString withFirefox ''firefox="${firefox}/bin/firefox"''}
-
-    log_depth="0"
-    args=""
-    browser=""
-
-    while [[ $# -ge 1 ]]; do
-      case "''${1}" in
-        -h|--help)
-          help
-          exit 0
-          ;;
-        -p|--profile)
-          if [[ -z "''${2:-}" ]]; then
-            echo -e "ERR: --profile requires an argument"
-            help
-            exit 1
-          fi
-          profile="''${2}"
-          shift 2
-          ;;
-        -b|--browser)
-          if [[ -z "''${2:-}" ]]; then
-            echo -e "ERR: --browser requires an argument"
-            help
-            exit 1
-          fi
-          browser="''${2}"
-          shift 2
-          ;;
-        *)
-          args=( ''${@} )
-          break
-          ;;
-      esac
-    done
-
-    # the profile paths per browser
-    ${optionalString withBrave ''BRAVE_PROFILES_PATH="''${HOME}/.config/brave/profiles"''}
-    ${optionalString withChromium ''CHROMIUM_PROFILES_PATH="''${HOME}/.config/chromium/profiles"''}
-    ${optionalString withFirefox ''FIREFOX_PROFILES_PATH="''${HOME}/.mozilla/firefox/profiles"''}
-
-    # make sure we have both profile and a browser.
-    if [[ -n "''${profile:-}" ]] && [[ -z "''${browser:-}" ]]; then
-      echo -e "ERR: the profile was set to ''${profile}, but the browser is empty. The browser is required with --profile, aborting..."
-      exit 1
-    fi
-
-    if [[ -z "''${profile:-}" ]]; then
-      items=""
-      sep=""
-    ${optionalString withBrave ''
-      if [[ -d "$BRAVE_PROFILES_PATH" ]]; then
-        for i in $BRAVE_PROFILES_PATH/*; do
-          if [[ -d "''${i}" ]]; then
-            items="''${items}''${sep}brave@$(basename "''${i}")"
-            sep="\n"
-          fi
-        done
-      fi
-    ''}
-    ${optionalString withChromium ''
-      if [[ -d "$CHROMIUM_PROFILES_PATH" ]]; then
-        for i in $CHROMIUM_PROFILES_PATH/*; do
-          if [[ -d "''${i}" ]]; then
-            items="''${items}''${sep}chromium@$(basename "''${i}")"
-            sep="\n"
-          fi
+  rbrowser = writeScriptBin "rbrowser"
+    (
+      let
+        items = builtins.concatStringsSep "\n" (map (browser: ''"${browser.name}@${browser.profile}"'') browsers);
+
+        declare_flags = builtins.concatStringsSep "\n" (map
```

```suggestion
        declareFlags = builtins.concatStringsSep "\n" (map
```
Maybe let's keep the same naming conventions everywhere?